hw/ip/ascon/data/ascon.hjson

```diff
@@ -0,0 +1,696 @@
+      package: "edn_pkg"
+    }
+    { struct:  "hw_key_req"
+      type:    "uni"
+      name:    "keymgr_key"
+      act:     "rcv"
+      package: "keymgr_pkg"
+    }
+  ],
+
+  registers: [
+##############################################################################
+# input key registers
+    { multireg: {
+      name: "KEY_SHARE0",
+      resval: "0",
+      desc: '''
+        Initial Key Registers Share 0.
+
+        The actual initial key corresponds to Initial Key Registers Share 0 XORed with Initial Key Registers Share 1.
+        Loaded into the internal Full Key register upon starting encryption/decryption.
+        All key registers (Share 0 and Share 1) must be written at least once when the key is changed.
+        The order in which the registers are updated does not matter.
+        Can only be updated when the Ascon unit is idle.
+        If the Ascon unit is non-idle, writes to these registers are ignored.
+        Upon reset, these registers are cleared with pseudo-random data.
+      '''
+      count: "NumRegsKey",
+      cname: "KEY_SHARE0",
+      swaccess: "wo",
+      hwaccess: "hrw",
+      hwext:    "true",
+      hwqe:     "true",
+      fields: [
+        { bits: "31:0", name: "key_share0", desc: "Initial Key Share 0" }
+      ],
+      }
+    },
+    { multireg: {
+      name: "KEY_SHARE1",
+      resval: "0",
+      desc: '''
+        Initial Key Registers Share 1.
+
+        The actual initial key corresponds to Initial Key Registers Share 0 XORed with Initial Key Registers Share 1.
+        Loaded into the internal Full Key register upon starting encryption/decryption.
+        All key registers (Share 0 and Share 1) must be written at least once when the key is changed.
+        The order in which the registers are updated does not matter.
+        Can only be updated when the Ascon unit is idle.
+        If the Ascon unit is non-idle, writes to these registers are ignored.
+        Upon reset, these registers are cleared with pseudo-random data.
+      '''
+      count: "NumRegsKey",
+      cname: "KEY_SHARE1",
+      swaccess: "wo",
+      hwaccess: "hrw",
+      hwext:    "true",
+      hwqe:     "true",
+      fields: [
+        { bits: "31:0", name: "key_share1", desc: "Initial Key Share 1" }
+      ],
+      }
+    },
+##############################################################################
+# input nonce registers
+    { multireg: {
+      name: "NONCE_SHARE0",
+      desc: '''
+        Input Nonce Register Share 0.
+        The actual data corresponds to Nonce Input Registers Share 0 XORed with Nonce Input Registers Share 1.
+        If SW does not want to provide the Nonce masked, it can simply set one share to all zeros and provide the unmasked nonce in the other share.
+        All nonce registers (Share0 and Share 1) must be written each time a new message is processed.
+        Ascon requires the nonce to be unique for each message. However there are no hardware checks to enforce this.
```

Comment:
Nit: line break after `message.` IMO, it's better to do one sentence per line as it eases future changes / diffs.